osbs/api.py

```diff
@@ -505,7 +505,8 @@ def _do_create_prod_build(self, git_uri, git_ref,
             info_url_format=self.build_conf.get_info_url_format(),
             artifacts_allowed_domains=self.build_conf.get_artifacts_allowed_domains(),
             low_priority_node_selector=self.build_conf.get_low_priority_node_selector(),
-            equal_labels=self.build_conf.get_equal_labels()
+            equal_labels=self.build_conf.get_equal_labels(),
+            node_selector_platforms=self.build_conf.get_node_selector_platform()
```

Comment:
Has to be passed in to build_request in _do_create_prod_build() because that's were build_request params are set.

I addressed this by altering get_node_selector_platform to take a platforms argument, and only returning node_selectors.platform whose platform matches one of the ones in platforms.